### PR TITLE
Fixes BloomFilter when linking BSK from multiple app targets

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/0/1205630353434502/f

macOS PR: https://github.com/duckduckgo/macos-browser/pull/1713
iOS PR: Since I'm merging against a feature branch, the iOS PR is currently not necessary.
What kind of version bump will this require?: None since this is aiming at a feature branch

## Description

Fixes some issues with `BloomFilterWrapper` when linking BSK in more than one App target.

## Steps to test this PR

See macOS PR for testing instructions.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -86,11 +87,16 @@ let package = Package(
                 "Bookmarks"
             ]
         ),
-         .target(
-            name: "BloomFilterWrapper",
+        .target(
+            name: "BloomFilterObjC",
```
Comment (diegoreymendez) [inline Package.swift:91]: The existing filter was renamed to ObjC.  This should ideally be kept private, as importing this directly would recreate the issue we had.

File: Package.swift [hunk 2]
```diff
@@ -86,11 +87,16 @@ let package = Package(
...
             dependencies: [
                 .product(name: "BloomFilter", package: "bloom_cpp")
             ]),
+        .target(
+            name: "BloomFilterWrapper",
```
Comment (diegoreymendez) [inline Package.swift:96]: This is the public wrapper in Swift.  It wraps our ObjC wrapper while keeping it private.

File: Sources/BloomFilterWrapper/BloomFilterWrapper.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public func add(entry: String) {
+        bloomFilter.add(entry)
+    }
+
+    public func contains(_ entry: String) -> Bool {
+        bloomFilter.contains(entry)
+    }
+}
```
Comment (diegoreymendez) [inline Sources/BloomFilterWrapper/BloomFilterWrapper.swift:40]: Very straightforward wrapper.  It just forwards the method calls.

File: Sources/BloomFilterWrapper/BloomFilterWrapper.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+@_implementationOnly import BloomFilterObjC
```
Comment (diegoreymendez) [inline Sources/BloomFilterWrapper/BloomFilterWrapper.swift:20]: Make sure that `BloomFilterObjC` is not exposed externally.

The underscore in `@_implementationOnly` means it's subject to future changes, so we may need to revisit this if the Swift specification changes, but it's really convenient to ensure we're not exposing the underlying statically linked library..

Comment (samsymons) [reply Sources/BloomFilterWrapper/BloomFilterWrapper.swift:20]: I think this is fine - `@_implementationOnly` is known to be used in some fairly major production SDKs, so I have a hard time seeing anything happen to it without a suitable replacement.

**Review by samsymons — APPROVED** [review]
LGTM, tested this via the associated macOS PR. Bloom Filter is working as expected, and I don't have any build failures. Nice work!

**Comment by diegoreymendez** [comment]
Since this is aiming a feature branch I'm moving ahead to merge.

**Pull request merged**